addons/crm/static/src/js/forecast/forecast_model_extension.js

```diff
@@ -0,0 +1,117 @@
+/** @odoo-module alias=crm.ForecastModelExtension **/
+import ActionModel from 'web.ActionModel';
+
+/**
+ * File intended for `web` (could be used in all kind of forecast configurations)
+ * 
+ * This file contains the logic behind a special "Forecast" filter.
+ * Any such filter should set the context key {forecast_filter: 1}
+ * Another context key must also be set for the view using this model extension:
+ * {forecast_field: "date_field_name"}, which represents the date/time field on which
+ * the forecast should be applied
+ * 
+ * The main purpose is to be able to modify the domain depending on the groupby granularity
+ * when the field is a `date/time` field and is the `forecast_field`. The domain should filter
+ * records from the period (granularity) containing "now", or those where the forecast_field
+ * is not set.
+ * example:
+ *  today:          2021-04-21
+ *  granularity:    month
+ *  field name:     date_field
+ *  -> the domain would be: ['&', ['date_field', '=', false], ['date_field', '>=', '2021-04-01']]
+ */
+class ForecastModelExtension extends ActionModel.Extension {
+
+    //---------------------------------------------------------------------
+    // Public
+    //---------------------------------------------------------------------
+
+    /**
+     * @override
+     * @returns {any}
+     */
+    get(property) {
+        switch (property) {
+            case "domain": return this.getDomain();
+        }
+    }
+
+    /**
+     * @override
+     */
+    prepareState() {
```

Comment:
This is an override that does not call super, is it intended ?

Reply:
yes, the original function is an abstract :
https://github.com/odoo/odoo/blob/4aa89f146e2d6c03df5803dbb196caf771269b35/addons/web/static/src/js/model.js#L102-L104

Reply:
Still unclear in my mind why we don't call super() even if its implementation is a dummy one.